src/sqlfluff/rules/L064.py

```diff
@@ -102,6 +105,20 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         ):
             return LintResult(memory=context.memory)
 
+        # This rule can also cover quoted literals that are partially templated.
+        # I.e. when the quotes characters are _not_ part of the template we can
+        # meanigfully apply this rule.
+        templated_raw_slices = context.functional.segment.raw_slices.select(
+            rsp.is_slice_type("templated")
+        )
+        for raw_slice in templated_raw_slices:
+            pos_marker = context.segment.pos_marker
+            # This is to make mypy happy. Probably there is a better way of doing it?
+            assert isinstance(pos_marker, PositionMarker)
```

Comment:
What you have is okay with me. Pretty sure I've done similar things. mypy is usually nice, but sometimes the workarounds are ugly.